src/webview_provider/queryResultPanel.ts

```diff
@@ -172,12 +177,49 @@ export class QueryResultPanel extends AltimateWebviewProvider {
           payload.query as string,
           payload.fn as Promise<QueryExecution>,
         );
+        break;
+      case "queryResultTab:render":
+        this.dbtTerminal.debug(
+          "queryResultTab:render",
+          "rendering query result tab",
+          payload,
+        );
+        this._queryTabData = payload.queryTabData;
+        const webviewPanel = window.createWebviewPanel(
+          QueryResultPanel.viewType,
+          "query_result_" + getFormattedDateTime(),
+          {
+            viewColumn: ViewColumn.Active,
+          },
+          { enableScripts: true, retainContextWhenHidden: true },
+        );
+        this._tabPanel = webviewPanel;
+        this.renderWebview(webviewPanel);
+
         break;
       default:
         super.onEvent({ command, payload });
     }
   }
 
+  private renderWebview(webview: WebviewPanel) {
+    this._webview = webview.webview;
+    this.renderTabWebviewView();
+  }
+
+  protected renderTabWebviewView() {
+    if (!this._webview) {
+      return;
+    }
+
+    this.setupTabWebviewHooks();
+
+    this._webview.html = this.getHtml(
```

Comment:
It is better to use `renderWebviewView`, which has few conditions to render different type of views based on configuration.

Reply:
Again, the setup time of tab and panel will be different